daphne/src/roles.rs

```diff
@@ -582,14 +581,44 @@ pub trait DapHelper<S>: DapAggregator<S> {
                     ));
                 }
 
+                // Remove reports that are rejected early.
+                let early_rejects = early_rejects_future.await?;
                 let agg_resp = match transition {
-                    DapHelperTransition::Continue(state, agg_resp) => {
+                    DapHelperTransition::Continue(state, mut agg_resp) => {
+                        // state.seq is a vector of VdafState and Nonce. To efficiently access and remove
+                        // unwanted VdafState, we convert the vector into a HashMap.
+                        //
+                        // TODO(nakatsuka-y) Change DapHelperState.seq from Vector to HashMap.
```

Comment:
Building this hash map is going to be expensive if the number of reports is large, say >100. We should be able to avoid this.

Consider the following algorithm for deleting `state`s (from rust docs):
```rust
let mut i = 0;
while i < vec.len() {
    if some_predicate(&mut vec[i]) {
        let val = vec.remove(i);
        // your code here
    } else {
        i += 1;
    }
}
```

Since `states.seq` and `agg_resp.transitions` are in 1:1 correspondence, it should be possible to extend this code to iterate over `agg_resp.transitions`.

Reply:
The updated code should reflect this.